develop/core/views.py

```diff
@@ -22,26 +27,29 @@ class ProductAccessView(ProductRequiredMixin, TemplateView):
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['object'] = self.model.on_site.get(slug=kwargs['slug'])
+        site = get_current_site(request)
+        context['object'] = self.model.objects.get(site=site, slug=kwargs['slug'])
         return render(request, self.template_name, context)
 
     def get_product_queryset(self):
         """
         Method to get the Product(s) needed for the check.  Can be overridden to handle complex queries.
         """
-        return self.model.on_site.filter(slug=self.kwargs['slug']).get().products.all()
+        site = get_current_site(request)
+        return self.model.objects.filter(site=site, slug=self.kwargs['slug']).get().products.all()
 
 class AccountView(LoginRequiredMixin, TemplateView):
     template_name = "core/account.html"
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-
-        customer_profile, created = request.user.customer_profile.get_or_create(site=Site.objects.get_current())
+        site = Site.objects.get_current()
```

Comment:
This will not work since get_current() uses the SITE_ID setting.  Get the site from the request and use this only if the attribute does not exist.

Reply:
Added SetSiteToRequestMixin